best-practices/tidb-partitioned-tables-best-practices.md

```diff
@@ -0,0 +1,703 @@
+
+To create a global index when creating a table, define the global index inline in the `CREATE TABLE` statement:
+
+```sql
+CREATE TABLE t (
+  id BIGINT NOT NULL,
+  col1 VARCHAR(50),
+  col2 VARCHAR(50),
+  -- other columns...
+  UNIQUE GLOBAL INDEX idx_col1_col2 (col1, col2)
+)
+PARTITION BY RANGE (id) (
+  PARTITION p0 VALUES LESS THAN (10000),
+  PARTITION p1 VALUES LESS THAN (20000),
+  PARTITION pMax VALUES LESS THAN MAXVALUE
+);
+```
+
+#### Summary
+
+The performance overhead of partitioned tables in TiDB depends significantly on the number of partitions and the type of index used.
+
+- The more partitions you have, the more severe the potential performance degradation.
+- With a smaller number of partitions, the impact might not be noticeable, but it is still workload-dependent.
+- For local indexes, if a query does not include effective partition pruning conditions, the number of partitions directly correlates with the number of [Remote Procedure Calls (RPCs)](https://docs.pingcap.com/tidb/stable/glossary/#remote-procedure-call-rpc) triggered. This means more partitions will likely result in more RPCs, leading to higher latency.
+- For global indexes, the number of RPCs and the degree of performance regression depend on both the number of partitions involved and how many rows need to be retrieved (that is, the number of rows requiring table lookups). Note that for very large tables where data is already distributed across many Regions, accessing data through a global index might have similar performance to a non-partitioned table, as both scenarios require multiple cross-Region RPCs.
+
+#### Recommendations
+
+- Do not use partitioned tables unless necessary. For most OLTP workloads, a well-indexed non-partitioned table performs better and is easier to manage.
+- If you are sure that all queries will make use of good partition pruning (matching only a few partitions), then local indexes are a good choice.
+- If you are sure that critical queries do not have good partition pruning (matching many partitions), then a global index is recommended.
+- Use local indexes only if your main concern is DDL efficiency (such as fast `DROP PARTITION`) and the performance side effect from the partition table is acceptable.
+
+## Facilitate bulk data deletion
+
+In TiDB, you can clear up historical data either by TTL (Time-to-Live) or manual partition drop. While both methods serve the same purpose, they differ significantly in performance. The test cases in this section show that dropping partitions is generally faster and less resource-intensive, making it a better choice for large datasets and frequent purging needs.
+
+### Differences between TTL and `DROP PARTITION`
+
+- TTL: automatically removes data based on its age, but might be slower due to the need to scan and clean data over time.
+- `DROP PARTITION`: deletes an entire partition at once, making it much faster, especially when dealing with large datasets.
```

Comment:
```suggestion
- TTL: automatically deletes data based on its age. This method might be slower because it scans and deletes rows incrementally over time.
- `DROP PARTITION`: deletes an entire partition in a single operation. This approach is typically much faster, especially for large datasets.
```